pandas/core/indexing.py

```diff
@@ -165,7 +166,28 @@ def _get_loc(self, key: int, axis: int):
     def _slice(self, obj, axis: int, kind=None):
         return self.obj._slice(obj, axis=axis, kind=kind)
 
+    def _ensure_listlike_indexer(self, key, axis):
```

Comment:
pls type these arguments & add a doc-string. make sure to indicate that this is a mutating operation.